### ISSUE TITLE
Cannot update a component (`Page`) while rendering a different component (`MuiPhone`).

### ISSUE THREAD

**Issue by beardsleym** [goveo/react-international-phone#51]
**Describe the bug**
Copying the MuiPhone example from the [docs](https://react-international-phone-docs.vercel.app/docs/Advanced%20Usage/useWithUiLibs)
There is a React Console Error.

> Cannot update a component (`Home`) while rendering a different component (`MuiPhone`). To locate the bad setState() call inside `MuiPhone`, 

**To Reproduce**
Copy the Material UI example Component from [here](https://react-international-phone-docs.vercel.app/docs/Advanced%20Usage/useWithUiLibs) and pass in a `value` and `onChange` function

``` ts
// home.tsx
const [phone, setPhone] = useState("+1");

  const handlePhoneValueChange = (value: string) => {
    setPhone(value);
  };
  
   <MuiPhone
        value={phone}
        onChange={handlePhoneValueChange}
      />
```
Open developer console in browser to see error

**Expected behavior**
Should not show errors

**Screenshots**

**Desktop (please complete the following information):**

- OS: [MacO]
- Browser [Chrom]
- Version [e.g. 112]

**Additional context** 

- https://d17c3b-3000.csb.app/ - see console errors
- https://codesandbox.io/p/sandbox/react-international-phone-d17c3b - code example

![Screenshot 2023-04-30 at 8 11 55 PM](https://user-images.githubusercontent.com/6046506/235386952-91782eaa-7559-4496-895f-2180733e0525.png)

**Comment by goveo** [comment]
@beardsleym Error is fixed in [v2.2.2](https://github.com/goveo/react-international-phone/releases/tag/v2.2.2). Thanks! 🎉

**Comment by beardsleym** [comment]
Awesome, well done. Thanks 🙏
